File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Decorator/Java/MilkDecorator.java
public class MilkDecorator extends CoffeeDecorator {

    public double milkCost = 1.5;

    public MilkDecorator(Coffee coffee) {
        super(coffee);
    }

    @Override
    public String getDescription() {
        return decoratedCoffee.getDescription() + ", Milk";
    }

    @Override
    public double getCost() {
        return decoratedCoffee.getCost() + milkCost;
    }
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Decorator/Cpp/src/MilkDecorator.cpp
#include "../include/MilkDecorator.h"

MilkDecorator::MilkDecorator(Coffee* coffee) : CoffeeDecorator(coffee) {}

std::string MilkDecorator::getDescription() const {
    return decoratedCoffee->getDescription() + ", Milk";
}

double MilkDecorator::getCost() const {
    return decoratedCoffee->getCost() + 1.5;
}
